

syntax = "proto3";

package glog;

import "paxos.proto";


message NoopMsg {
};


message ProposeRequest {
    bytes data = 1;
};

message ProposeResponse {
    int32 retcode = 1;
};

message GetGlogRequest {
    uint64 index = 1;
};

message GetGlogResponse {
    string info = 1;
    bytes data = 2;
};

message PaxosInfo {
    uint64 max_index = 1;
    uint64 commited_index = 2;
};

service Glog {

    rpc PostMsg (paxos.Message) returns (NoopMsg) {}

    rpc Propose (ProposeRequest) returns (ProposeResponse) {}

    // internal use
    rpc GetPaxosInfo (NoopMsg) returns (PaxosInfo) {}

    rpc TryCatchUp (NoopMsg) returns (NoopMsg) {}

    // ADD FOR TEST
    rpc GetGlog (GetGlogRequest) returns (GetGlogResponse) {}

}




